### PR TITLE
Fix for issues #187 and #146

### LINKED ISSUE #146: IgTreeGridComponent is throwing an error when there is no dataSource specified
Based on forum thread [http://www.infragistics.com/community/forums/t/109424.aspx](http://www.infragistics.com/community/forums/t/109424.aspx)

### LINKED ISSUE #187: Cannot set remote url for igHierarchicalGrid data source
### Description  
When setting remote url: "https://services.odata.org/V2/Northwind/Northwind.svc/Regions" as data source for igHierarchicalGrid an error is thrown.

### Steps to reproduce  

1. Open the attached sample
2. Run npm install
3. Run npm start
4. Observe the browser
[ngHGrid_UsingOdata.zip](https://github.com/IgniteUI/igniteui-angular2/files/1215055/ngHGrid_UsingOdata.zip)

 console

### Result  
ERROR Error: Error trying to diff 'https://services.odata.org/V2/Northwind/Northwind.svc/Regions'. Only arrays and iterables are allowed
    at DefaultIterableDiffer.diff (core.umd.js:6862)
    at IgHierarchicalGridComponent.execute.IgHierarchicalGridComponent.ngDoCheck (igniteui.angular2.ts!transpiled:2328)
    at checkAndUpdateDirectiveInline (core.umd.js:10871)
    at checkAndUpdateNodeInline (core.umd.js:12364)
    at checkAndUpdateNode (core.umd.js:12303)
    at debugCheckAndUpdateNode (core.umd.js:13164)
    at debugCheckDirectivesFn (core.umd.js:13105)
    at Object.eval [as updateDirectives] (igHierarchicalGridTemplate.html:3)
    at Object.debugUpdateDirectives [as updateDirectives] (core.umd.js:13090)
    at checkAndUpdateView (core.umd.js:12270)

### Expected result  
No error should be thrown and the grid should bind to the remote url without two-way data binding functionality.

### Attachments  
Attach a sample if available, and screenshots, if applicable.

### PR DESCRIPTION
**Pull request by tbelcheva**
Closes #187 
Closes #146

### CODE REVIEW

**Comment by MayaKirova** [comment]
Two-way databinding tests are failing. Probably because the return condition is wrong.
```
if (typeof this._config.dataSource === "string" || this._config.dataSource instanceof Array) {
return; 
}
```
We should return if the data source is not array. If it is the two-way databinding logic should execute as before.
Example:
```
if( !(this._config.dataSource instanceof Array) ) {
    return;
 }
this._changes = this._differ.diff(this._config.dataSource);
```

**Comment by kdinev** [comment]
@tbelcheva Can you include some tests for this scenario?

**Comment by kdinev** [comment]
@tbelcheva `bower_components/` should be added to the `.gitignore`

**Comment by mpavlinov** [comment]
@tbelcheva Please use https://www.npmjs.com/package/jquery-mockjax instead of bower

File: tests/unit/iggrid/grid.spec.ts [hunk 1]
```diff
@@ -719,6 +742,7 @@ class TestComponent {
 	public opts2: any;
 	public opts3: any;
 	public optsNew:any;
+	public opts4: any;
```
Comment (mpavlinov) [inline tests/unit/iggrid/grid.spec.ts]: @tbelcheva Where do you configure opts4?

File: tests/unit/igtreegrid/treegrid.spec.ts [hunk 2]
```diff
@@ -159,13 +182,13 @@ export function main() {
...
     });
 }
-
 @Component({
     selector: 'test-cmp',
     template: '<div></div>' //"Component 'TestComponent' must have either 'template' or 'templateUrl' set."
 })
 class TestComponent {
     private opts: any;
+    private opts2: any;
```
Comment (mpavlinov) [inline tests/unit/igtreegrid/treegrid.spec.ts]: @tbelcheva Where do you configure opts2?

**Pull request merged**